components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java

```diff
@@ -2239,4 +2258,501 @@ public static ClusterBean getClusterInfo(String clusterId) throws RestAPIExcepti
 
         return ObjectConverter.convertClusterToClusterBean(cluster, clusterId);
     }
+
+    //util methods for Tenants
+
+    /**
+     * Add Tenant
+     *
+     * @param tenantInfoBean
+     * @throws RestAPIException
+     */
+    public static void addTenant(org.apache.stratos.common.beans.TenantInfoBean tenantInfoBean) throws RestAPIException {
+
+        try {
+            CommonUtil.validateEmail(tenantInfoBean.getEmail());
+        } catch (Exception e) {
+            String msg = "Invalid email is provided";
+            log.error(msg, e);
+            throw new RestAPIException(msg);
+        }
+
+        String tenantDomain = tenantInfoBean.getTenantDomain();
+        try {
+            TenantMgtUtil.validateDomain(tenantDomain);
+        } catch (Exception e) {
+            String msg = "Tenant domain validation error for tenant " + tenantDomain;
+            log.error(msg, e);
+            throw new RestAPIException(msg);
+        }
+
+        UserRegistry userRegistry = (UserRegistry) PrivilegedCarbonContext.getThreadLocalCarbonContext().
+                getRegistry(RegistryType.USER_GOVERNANCE);
+        if (userRegistry == null) {
+            log.error("Security alert! User registry is null. A user is trying create a tenant "
```

Comment:
Log and throw is generally considered a bad practice. If not extremely necessary, we should probably be able to get rid of it in this sort of instances.